modules/ProjectScore.py

```diff
@@ -49,6 +50,45 @@ def generate_project_score_evidence(
     )
 
 
+def get_disease_cell_lines(
+    cell_passport_data: GenerateDiseaseCellLines, cell_line_data: DataFrame
+) -> DataFrame:
+    """Based on the cell passport data and cell line data, generate a dataframe with disease cell lines.
+
+    Args:
+        cell_passport_data (GenerateDiseaseCellLines): Cell passport data as a GenerateDiseaseCellLines object
+        cell_line_data (DataFrame): Cell line data from ProjectScore as a DataFrame
+
+    Returns:
+        DataFrame: Prepared disease cell lines objects indexed as cancerType.
+    """
+    passport_disease_cell_lines = cell_passport_data.generate_disease_cell_lines()
+
+    # Joining disease cell-lines dataframe for Project Score:
+    disease_cell_lines = (
+        cell_line_data.select(
+            f.lower(f.col("CANCER_TYPE")).alias("cancerType"),
+            f.col("CMP_ID").alias("id"),
+        )
+        .join(passport_disease_cell_lines, on="id", how="right")
+        .groupBy("cancerType")
+        .agg(f.collect_set("diseaseCellLine").alias("diseaseCellLines"))
+    )
+    # Get report:
+    logger = logging.getLogger(__name__)
```

Comment:
I'd suggest using the same logger we set in `main`, instead of initialising a new one.
To do that, I see 3 easy options:
1.  either return `disease_cell_lines` here and move the reporting logic to main (my preferred option because of separation of concerns)
2. pass the logger as an argument of `get_disease_cell_lines`
3. in main, make `logger` a global variable
```
global logger
logger = logging.getLogger(__name__)
```

Reply:
In my understanding, that statement instead of initializing the logger, it retrieves the logger that was already been formatted for the module, as `__name__` is given, which is module-wide identifier. Also used identically in the main function as well (line 111). I'm not clear on the details, but as seen on the logs, messages from this function is definitely formatted consistently (implies that they are the same loggers), also true if `--log_file` is specified:

```
2024-03-12 12:50:00 INFO ProjectScore - main: Running Project Score parser with parameters:
2024-03-12 12:50:00 INFO ProjectScore - main: Evidence file: /Users/dsuveges/Downloads/project_score_v2/mapped_diseases.tsv
2024-03-12 12:50:00 INFO ProjectScore - main: Cell types file: /Users/dsuveges/Downloads/project_score_v2/cell_types.tsv
2024-03-12 12:50:00 INFO ProjectScore - main: Cell passport file: /Users/dsuveges/Downloads/model_list_20240110.csv
2024-03-12 12:50:00 INFO ProjectScore - main: Cell line to uberon mapping: depmap_uberon_mapping.csv
2024-03-12 12:50:00 INFO ProjectScore - main: Output file: projectscore_v2b.json.gz
2024-03-12 12:50:05 INFO ProjectScore - get_disease_cell_lines: All cancer types are in the cell line file.
2024-03-12 12:50:06 INFO ProjectScore - main: Number of evidence strings: 517
2024-03-12 12:50:07 INFO ProjectScore - main: Number of targets in evidence data: 302
2024-03-12 12:50:07 INFO ProjectScore - main: Number of diseases in evidence data: 26
2024-03-12 12:50:07 INFO ProjectScore - main: Writing evidence strings to file...
```